src/main/java/vn/unigap/api/controller/EmployerController.java

```diff
@@ -0,0 +1,114 @@
+package vn.unigap.api.controller;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import vn.unigap.api.dto.in.EmployerCreateRequestDTO;
+import vn.unigap.api.dto.in.EmployerUpdateRequestDTO;
+import vn.unigap.api.dto.out.ApiResponseDTO;
+import vn.unigap.api.dto.out.EmployerResponseDTO;
+import vn.unigap.api.entity.Employer;
+import vn.unigap.api.service.EmployerService;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/employers")
+public class EmployerController {
+    private final EmployerService employerService;
+
+    @Autowired
+    public EmployerController(EmployerService employerService) {
+        this.employerService = employerService;
+    }
+
+    @PostMapping("/add")
+    public ResponseEntity<ApiResponseDTO> createEmployer(@RequestBody EmployerCreateRequestDTO requestCreateDTO) {
+        // Create the employer
+        try {
+            Employer employer = employerService.createEmployer(requestCreateDTO);
+            return ResponseEntity.status(HttpStatus.CREATED)
+                    .body(new ApiResponseDTO(0, 201, "Employer created successfully", employer));
+
+        } catch (Exception e) {
+            return toApiResponse(e);
+        }
+    }
+
+    private static ResponseEntity<ApiResponseDTO> toApiResponse(Exception e) {
```

Comment:
Should consider to move this function to a common class. May be is a abstract class or a common util class

Reply:
Done. Move to api/exception/Exceptionhandler.java